Assets/Scripts/AI/Entities/Colonist.cs

```diff
@@ -1,19 +1,44 @@
 using UnityEngine;
+using WorldAPI.Tasks.Generic;
 
-public class Colonist : AIEntity
+public partial class Colonist : ResourceGatherer<Colonist>
 {
-    public void Start()
+    public new void Start()
     {
-        Init(
-            health: 10 + Random.Range(0, 5),
-            damage: 1 + Random.Range(0, 3)
-            );
+        base.Start();
 
-        AssignTask(new GatherResourceTask(this, Material.Stone));
+        Health = 10 + Random.Range(0, 5);
+        MaxHealth = Health;
+
+        Damage = 1 + Random.Range(0, 3);
+
+        QueueTask(new IdleTask<Colonist>());
     }
 
     public new void Update()
     {
         base.Update();
     }
 }
+
+public partial class Colonist { 
+    private static GameObject _prefab;
+
+    public static Colonist New(string name = "Colonist", Vector3? location = null, Faction faction = null)
+    {
+        return Derive(name, location, faction).GetComponent<Colonist>();
+    }
+
+    public static GameObject Derive(string name = "Colonist", Vector3? location = null, Faction faction = null)
```

Comment:
Why do you call this function Derive?

Reply:
I don't think it really matters in this case and `New` and `Derive` can be merged, but my thinking was kind of an inheritance type-system where if Colonist extends on say `GameEntity`, it will "derive" from a gameobject with all the necessary components for a `GameEntity`. Then, the New() method would take this base, and add the required MonoBehaviour element and return it. Since Colonist uses a prefab, it already has all the predefined components needed, it won't need to derive from a superclass. And due to the fact that Colonist probably won't be subclassed, Derive is probably pointless in this case.

Reply:
Fixed.